src/blockdata/script.rs

```diff
@@ -1107,6 +1107,47 @@ impl Decodable for Script {
     }
 }
 
+/// Search for `needle` in the vector `haystack` and remove every
+/// instance of it, returning the number of instances removed.
+/// Loops through the vector opcode by opcode, skipping pushed data.
+pub fn script_find_and_remove(haystack: &mut Vec<u8>, needle: &[u8]) -> usize {
```

Comment:
Where is this used? Do we even need it? It's unclear to me what "skipping pushed data" even means.
Given this is intended to operate on script I would prefer this to be a method on `Script`.
The code also looks like it may be able to use `Vec::retain`

Reply:
This is a copy of https://github.com/bitcoin/bitcoin/blob/master/src/script/interpreter.cpp#L228-L254 . It cannot use `Vec::retain` because it needs to distinguish between pushed data and script opcodes.

I think we should probably deprecate this and remove it. Its only use is in computing sighashes in cases that users are doing pretty obscure things (using `CODESEPARATOR` and/or using pubkey-recovery hacks to force an ECDSA signature to sign itself) but then not using Segwit. I think at this point that's niche enough that we can forget about it.

Reply:
If we do keep it it should probably go into the `signature` module where it was before.

Reply:
I'll put it back in `signature` and deprecate it.

Reply:
Do we ever plan on supporting `CODESEPARATOR`? We have a bunch of places in sighash code where we explicitly state that we are not handling it. If we are we should keep this then, right?

Reply:
I think if we ever decide to support `CODESEPARATOR` we can just put it back.

@apoelstra `retain` takes `FnMut` so it's possible to have a state tracking whether we're handling push byte or not. But maybe the result would be worse than what we have now. Anyway, I agree with removing, so this is just an academic discussion. :)

Reply:
Hah, that's a neat idea maintaining "are we inside a push" state. It probably wouldn't be too bad, actually. It just wouldn't have occurred to me because I was still mentally transliterating "C with better enums" to Rust back when I did this function.

But yes, it's academic :)

And +1 to "if we decide to support `CODESEPARATOR` we can put tit back", but also, if we *did*, we'd probably support the Taproot-style `CODESEPARATOR` instead which is far far easier to reason about. (It just pushes its own index into the sighash rather than attempting to mangle other hashed data.)